### DISCUSSION TITLE
adding the plugin locally

### DISCUSSION THREAD

**Discussion by BellBoy68** [Budibase/budibase#10269] [Help] [answered]
Hi I am trying to add the following plugin to my budibase:
https://github.com/melohagan/budibase-component-nikita-input-mask.git

An error occurs when trying to add:
![image](https://user-images.githubusercontent.com/127862266/231372018-eb1dfcfc-be74-4282-86a2-b7745b939575.png)

What to do?

**Comment by andz-bb** [comment]
Hey @BellBoy68, for the GitHub plugin import you just need to paste in the regular web URL for the repository rather than the api URL. Can you try pasting in `https://github.com/melohagan/budibase-component-nikita-input-mask` instead?

**Reply by BellBoy68** [reply]
> https://github.com/melohagan/budibase-component-nikita-input-mask

With this link shows the same error

**Reply by andz-bb** [reply]
@BellBoy68 can you send a screenshot of the error or show me the error in the browser's developer tools console please? I just successfully imported the plugin on both my cloud and self host tenants using that URL. Could you also describe your hosting setup as it's possible your instance is failing to connect to the internet

**Reply by BellBoy68** [reply]
Error in console:
![image](https://user-images.githubusercontent.com/127862266/231420342-e7532d52-e093-4c6c-8ae0-afb04c0b4a3d.png)

My budibase is running locally from docker-compose file. The file was taken from the budibase website.

**Reply by ConorWebb96** [reply]
Hey @BellBoy68, 

Would you be able to provide a video or gif of you going through the process to add a plugin?
Could you also let me know what version of Budibase you are running?

Can you just confirm you are following a similar process to what I'm showing below?
![adding custom plugin](https://user-images.githubusercontent.com/126772285/231435183-1cff3a26-e521-4a65-8f04-cd75050378fa.gif)

You could also try the file upload version, this may work for you instead. All you would need to do is clone down the plugin and upload it instead.
<img width="518" alt="Screenshot 2023-04-12 at 11 50 02" src="https://user-images.githubusercontent.com/126772285/231436059-b1b62824-41b3-4ea9-b4ab-8b065ad50690.png">

Hopefully we can get this sorted for you!

**Reply by BellBoy68** [reply]
Yes, I follow this process as in the video. Budibase version 2.3.19.
When trying to download
tar file the following error appears:
![image](https://user-images.githubusercontent.com/127862266/231440042-8fff6c27-a7de-4ce2-afe5-b6ccaae95ef7.png)

Maybe it's a proxy server. If so, what should be written in docker-compose?

**Reply by ConorWebb96** [reply]
@BellBoy68, 

Can you update to the latest version as you are a bit behind?
If this still doesn't work let us and we will look into your proxy suggestion within docker-compose.

Thanks!

**Reply by BellBoy68** [reply]
I updated budibase to version 2.5.2. But I still can't load the plugin.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @BellBoy68,

Can you try uploading with the below? I think the issue with the upload you tried previously was down to the folder not being compressed.
[nikita-input-mask-1.0.0.tar.gz](https://github.com/Budibase/budibase/files/11211043/nikita-input-mask-1.0.0.tar.gz)

Have you made amends or modifications to the docker compose file? Do you have an active VPN on that may be cause these network issues?

Hopefully this helps!

**Reply by BellBoy68** [reply]
Thank you. Indeed, the problem is in the incorrectly archived file. The plugin has now loaded. I didn't change the configuration.
